backend/tests/models/user.test.ts

```diff
@@ -29,168 +28,404 @@ describe('User model and validation tests', () => {
     expect(value).toStrictEqual(testObj);
   });
 
-  it('returns error if email over 100 chars', () => {
-    const testObj = {};
-    const testOptions = {};
-    
-    const { error, value } = User.baseJoi(testObj, testOptions);
-
-    expect(error).toBeUndefined();
-    expect(value).toBeNull();
-  });
+  const invalidEmails = [
+    { desc: 'email over 100 chars', email: 'A'.repeat(101) + '@fakesite.com' },
+    { desc: 'email uses capital letters', email: 'TestEmail@fakesite.com' },
+    { desc: 'email extension is not allowed', email: 'testemail@fakesite.xyz' },
+  ];
 
-  it('returns error if email uses capital letters', () => {
-    const testObj = {};
+  it.each(invalidEmails)('returns error if %s', ({ email }) => {
+    const testObj = { email };
     const testOptions = {};
     
-    const { error, value } = User.baseJoi(testObj, testOptions);
+    const { error } = User.baseJoi(testObj, testOptions);
 
-    expect(error).toBeUndefined();
-    expect(value).toBeNull();
+    expect(error).not.toBeNull();
```

Comment:
I think there's an issue here. You'd want to use toBeDefined() or toBeTruthy() instead of not.toBeNull() here because if typeof error === typeof undefined it would still pass this test because undefined !== null. My understanding is that if validation succeeds, error is undefined in the result, while if validation fails, error will be set.
```suggestion
    expect(error).toBeDefined();
```